ecl/hql/hqldesc.cpp

```diff
@@ -136,35 +235,42 @@ void expandSymbolMeta(IPropertyTree * metaTree, IHqlExpression * expr)
         IHqlNamedAnnotation * symbol = queryNameAnnotation(expr);
         def->setProp("@name", str(expr->queryId()));
         def->setPropInt("@line", expr->getStartLine());
+        setFullNameProp(def, "@fullname", expr);
+
         if (expr->isExported())
             def->setPropBool("@exported", true);
         else if (isPublicSymbol(expr))
             def->setPropBool("@shared", true);
 
+        def->setProp("@inherit_type", getInheritTypeText(ihType));
+
         if (symbol)
         {
-            setNonZeroPropInt(def, "@start", symbol->getStartPos());
-            setNonZeroPropInt(def, "@body", symbol->getBodyPos());
-            setNonZeroPropInt(def, "@end", symbol->getEndPos());
+            def->setPropInt("@start", symbol->getStartPos());
+            def->setPropInt("@body", symbol->getBodyPos());
+            def->setPropInt("@end", symbol->getEndPos());
+            def->setProp("@source", str(expr->querySourcePath()));
```

Comment:
we use sourcepath to read ecl file and extract signature for corresponding definition from it. While most definitions has source path of containing tag, some (specially inherited and overriden attributes) do have sources in other files. 
What I can do is only add source path only if it different from one in containing tag.